container-chains/templates/simple/node/src/command.rs

```diff
@@ -282,6 +284,43 @@ pub fn run() -> Result<()> {
                 ))
             })
         }
+        Some(Subcommand::RpcProvider(cmd)) => {
+            let runner = cli.create_runner(&cli.run.normalize())?;
+
+            runner.run_node_until_exit(|_config| async move {
+                let client: Box<dyn OrchestratorChainInterface>;
+                let mut task_manager;
+
+                if cmd.orchestrator_endpoints.is_empty() {
+                    todo!("Start in process node")
+                } else {
+                    task_manager = TaskManager::new(tokio::runtime::Handle::current(), None)
+                        .map_err(|e| sc_cli::Error::Application(Box::new(e)))?;
+
+                    client = dc_orchestrator_chain_rpc_interface::create_client_and_start_worker(
+                        cmd.orchestrator_endpoints.clone(),
+                        &mut task_manager,
+                        None,
+                    )
+                    .await
+                    .map(Box::new)
+                    .map_err(|e| sc_cli::Error::Application(Box::new(e)))?;
+                };
+
+                // POC: Try to fetch some data through the interface.
+                task_manager
+                    .spawn_handle()
+                    .spawn("rpc_provider_exemple", None, async move {
+                        let mut stream = client.new_best_notification_stream().await.unwrap();
+
+                        while let Some(header) = stream.next().await {
+                            println!("New best block: {}", header.hash());
```

Comment:
```suggestion
                            log::info!("New best block: {}", header.hash());
```

Reply:
this is where, for every new header, we would ask whether we have being assigned to a new chain right?